ocaml/xapi/xapi_vm.ml

```diff
@@ -1651,13 +1651,20 @@ let get_secureboot_readiness ~__context ~self =
               Helpers.call_script !Xapi_globs.varstore_ls
                 [Db.VM.get_uuid ~__context ~self]
             in
-            let contains str substr =
-              Astring.String.is_infix ~affix:substr str
+            let ls_lines = String.split_on_char '\n' varstore_ls in
+            let ls_keys =
+              List.map
+                (fun elem ->
+                  (* Lines follow this pattern: <GUID> <KEY>*)
+                  let splitted = String.split_on_char ' ' elem in
+                  List.nth splitted 1
```

Comment:
Nth can raise with a pretty nasty error that's difficult to find where it originated. Please use nth_opt and deal with the option by replacing List.map with List.filter_map

Reply:
Done :+1: